server/bundler/loader.js

```diff
@@ -66,6 +66,7 @@ function splitTemplate( path, module, chunkName ) {
 
 	result = [
 		'page( ' + path + ', function( context, next ) {',
+		'	context.sectionRouteMatched = true;',
```

Comment:
This approach will only allow us to catch 404s on paths defined in `sections.js`, so we wouldn't catch something like `/me/foo`. I wonder if there's a clean way of catching unmatched routes after the `require()` in `require.ensure`.